api/upstream.go

```diff
@@ -0,0 +1,54 @@
+package api
+
+import (
+	"strings"
+	"time"
+
+	"github.com/flanksource/duty/models"
+)
+
+// PushData consists of data about changes to
+// components, configs, analysis.
+type PushData struct {
+	CheckedAt                    time.Time `json:"checked_at"`
+	Labels                       map[string]string
+	Canaries                     []models.Canary
+	Checks                       []models.Check
+	Components                   []models.Component
+	ConfigAnalysis               []models.ConfigAnalysis
+	ConfigChanges                []models.ConfigChange
+	ConfigItems                  []models.ConfigItem
+	CheckStatuses                []models.CheckStatus
+	ConfigRelationships          []models.ConfigRelationship
+	ComponentRelationships       []models.ComponentRelationship
+	ConfigComponentRelationships []models.ConfigComponentRelationship
+}
+
+type UpstreamConfig struct {
+	URL      string
+	Username string
+	Password string
+	Labels   []string
+}
+
+func (t *UpstreamConfig) IsFilled() bool {
+	return t.URL != "" && t.Username != "" && t.Password != ""
+}
+
+func (t *UpstreamConfig) LabelsMap() map[string]string {
+	return sanitizeStringSliceVar(t.Labels)
+}
+
+func sanitizeStringSliceVar(in []string) map[string]string {
```

Comment:
Can  be moved to utils package

Reply:
@moshloop Should we move this to commons repo ? duty and canary checker also have a similar function ie. parsing raw labels into map[string]string

Reply:
Yeh, makes sense

Reply:
Added https://github.com/flanksource/commons/pull/58